iib/workers/tasks/fbc_utils.py

```diff
@@ -118,3 +123,23 @@ def extract_fbc_fragment(temp_dir: str, fbc_fragment: str) -> Tuple[str, str]:
         raise IIBError("More than 1 package is present in fbc_fragment %s", fbc_fragment)
 
     return fbc_fragment_path, operator_packages[0]
+
+
+def enforce_json_config_dir(config_dir: str) -> None:
+    """
+    Ensure the files from config dir are in JSON format.
+
+    It will walk recursively and convert any YAML files to the JSON format.
+
+    :param str config_dir: The config dir to walk recursively converting any YAML to JSON.
+    """
+    for dirpath, _, filenames in os.walk(config_dir):
+        for file in filenames:
+            in_file = os.path.join(dirpath, file)
+            if in_file.lower().endswith(".yaml"):
+                out_file = os.path.join(dirpath, f"{Path(in_file).stem}.json")
+                log.debug(f"Converting {in_file} to {out_file}.")
+                with open(in_file, 'r') as yaml_in, open(out_file, 'w') as json_out:
+                    data = yaml.load(yaml_in)
+                    json.dump(data, json_out)
```

Comment:
I would love to see some validation that converted output is correct.

Reply:
There's a validation in `tests/test_workers/test_tasks/test_fbc_utils.py:L175` already, do you mean adding an extra validation layer within the `fbc_utils.py` as well? Like, reopening the written data and ensuring it has the same data and if not raise IIBError?

Reply:
Oh, I get it now, `opm validate`, I'll implement it

Reply:
Added in abbc6e6452762c5810c40dfc5d2d783abc2e8b10